src/models/neural.ts

```diff
@@ -0,0 +1,137 @@
+import RestfulModel from './restful-model';
+import NeuralSentimentAnalysis from './neural-sentiment-analysis';
+import NeuralSignatureExtraction from './neural-signature-extraction';
+import NeuralOcr from './neural-ocr';
+import NeuralCategorizer, { Categorize } from './neural-categorizer';
+import NeuralCleanConversation from './neural-clean-conversation';
+import NeuralSignatureContact from './neural-signature-contact';
+
+export type NeuralMessageOptions = {
+  ignore_links?: boolean;
+  ignore_images?: boolean;
+  ignore_tables?: boolean;
+  remove_conclusion_phrases?: boolean;
+  images_as_markdowns?: boolean;
+};
+
+export default class Neural extends RestfulModel {
+  sentimentAnalysisMessage(
+    messageId: string
+  ): Promise<NeuralSentimentAnalysis> {
+    const body = { message_id: [messageId] };
+    return this._sentimentAnalysis(body);
+  }
+
+  sentimentAnalysisText(text: string): Promise<NeuralSentimentAnalysis> {
+    const body = { text: text };
+    return this._sentimentAnalysis(body);
+  }
+
+  extractSignature(
+    messageId: string,
+    parseContact?: boolean,
+    options?: NeuralMessageOptions
+  ): Promise<NeuralSignatureExtraction> {
+    let body: { [key: string]: any } = { message_id: [messageId] };
+    const path = 'signature';
+
+    if (options) {
+      body = {
+        ...body,
+        ...options,
+      };
+    }
+    if (parseContact) {
+      body['parse_contact'] = parseContact;
+    }
+
+    return this._request(path, body).then(json => {
+      if (Array.isArray(json)) {
```

Comment:
in which cases there may be more elements?

Reply:
According to the docs, never, but sometimes the API response that gets returned is in the form of an Array of just 1  element. Here's an example: https://developer.nylas.com/docs/intelligence/signature-extraction/#parse-signature-response